rust/otap-dataflow/crates/otap/src/parquet_exporter/schema.rs

```diff
@@ -0,0 +1,926 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains code for manipulating the arrow record batch [`RecordBatch`] so that it
+//! has a schema that's compatible with the Parquet writer.
+//!
+//! The main reason we can't write the record batch verbatim is because OTAP considers some columns
+//! optional, but parquet spec requires that each row group contain a column chunk for every field
+//! in the schema (and the column chunks must be in the correct order).
+//!
+//! This means we can't receive two consecutive OTAP batches for some payload type and write them
+//! into the same writer. To  handle this, we insert all null columns for missing columns (or all
+//! default-value where the column is not nullable), and also arrange the columns so they're always
+//! in the same order.
```

Comment:
We probably don't have any other simple option to implement to resolve this limitation. However, it's important to keep an eye on the implications this will have, particularly on memory consumption across the parquet -> arrow -> query engine path. It might have no effect, but we'll need to verify that.

Reply:
wrt to querying, I think the behvaiour will depend on the query engine but we can definitely verify. I imagine that generally the overhead of the extra metadata from these additional columns would be small compared to the size of the data. Depending on the query, we'd still probably have empty arrays in memory for these columns if the user selected them.